woocommerce/woocommerce-ios
# AddOns: Store product metadata

fix #3886 

# Why
This PR makes sure that the product add-ons parsed on #3942 are safely stored in our `Storage` layer as well as linked to a product when synchronizing/fetching products via the `ProductStore`.

# How
- Added Model-V47 with CoreData classes for `ProductAddOn` and `ProductAddOnOption`.
- Add CoreData `addOns` relationship on `Storage.Product`
- Make `ProductAddOn` and `ProductAddOnOption` conform to `ReadOnlyConvertible`
- Update `ProductStore` to persist addOns and addOns-options when `upserting` a product.
- Update `ProductStoreTests` to make sure everything is being stored correctly.
- Change the `imageID` type from `Int64` to `String`

# Testing steps

**Prerequisites:** Have your site with the add ons plugin installed, configured, and with at least one product with add-ons.

Add the following code on `ProductFormViewModel.init(...)` method

```swift
print("--------------- Add Ons ---------------")
for addOn in product.product.addOns {
  print("name: \(addOn.name)")
  print("options: \(addOn.options.map { $0.label })")
}
print("--------------- Add Ons ---------------")
```

- Launch the app 
- Navigate to the product that has addOns configured.
- See console printing addOns and options.

Update release notes:
